src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -375,6 +402,13 @@ private async Task<ResourceConfig<VirtualMachineScaleSet>> SimpleParameterSetNor
                         _cmdlet.WriteInformation(ValidateBase64EncodedString.UserDataEncodeNotification, new string[] { "PSHOST" });
                     }
                 }
+                
+                if (_cmdlet.IsParameterBound(c => c.SecurityType) && (_cmdlet.SecurityType == "TrustedLaunch" || _cmdlet.SecurityType == "ConfidentialVM"))
+                {
+                    _cmdlet.SecurityType = _cmdlet.SecurityType;
+                    _cmdlet.EnableVtpm = _cmdlet.EnableVtpm == null ? true : _cmdlet.EnableVtpm;
+                    _cmdlet.EnableSecureBoot = _cmdlet.EnableSecureBoot == null ? true : _cmdlet.EnableSecureBoot;
```

Comment:
This can be shortened to `_cmdlet.EnableSecureBoot = _cmdlet.EnableSecureBoot ?? true;`